core/templates/dev/head/pages/exploration_player/information_card_modal_directive.html

```diff
@@ -8,9 +8,10 @@ <h2 class="oppia-info-card-title"><[explorationTitle]></h2>
     <p ng-if="objective">
       <[objective | truncateAndCapitalize: 95]>
     </p>
-    <p ng-if="!objective" translate="I18N_PLAYER_NO_OBJECTIVE">No objective specified.</p>
+    <p ng-if="!objective && status == 'public'" translate="I18N_PLAYER_NO_OBJECTIVE">No objective specified.</p>
```

Comment:
'public' should be a named constant instead. You can define a method on the $scope along the lines of $scope.isExplorationPublic().

Reply:
Yeah! It didn't strike me!